### PR TITLE
Make return date mandatory and transfer out date and location required if patient is transferred out

### PR DESCRIPTION
**Pull request by semujju**
https://app.asana.com/0/1133167201254915/1201769013495240/f

Changes Made
Made return date required in both POC and Retrospective
Made transfer date and transfer location required if transfer out is checked both in POC and Retrospective

NB: 
- Created a return date tab in POC and made it active by default so that its easily visible since its required before saving

### CODE REVIEW

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 1]
```diff
@@ -1191,16 +1188,33 @@
                 var appointment = new Date((jq("#returndate").find('input[type=hidden]').val()));
                 var transferOutDate = new Date((getValue('transfer_out_date.value')));
 
+                if(getValue('returndate.value')==""){
+                    jq().toastmessage('showErrorToast', "The return date is required");
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @semujju The message is "The return visit date is required" - however why do you not just put a required attribute on the field

Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: When I used required="required", it wasn`t returning a message below the box that is why I used js. 
However, I have tested with required=true and it solves the problem. So I will remove the js

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 2]
```diff
@@ -1186,23 +1213,50 @@
                 setValue('other-meds-quantity.error','');
             </restrictByRole>
 
-
                 var encounterDate = new Date((getValue('encounterDate.value')));
                 var appointment = new Date((jq("#returndate").find('input[type=hidden]').val()));
-                var transferOutDate = new Date((getValue('transfer_out_date.value')));
+                var transferOutDate = new Date((getValue('transfer_out_date.value')));
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1218]: @semujju I see extra parentheses in here

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 3]
```diff
@@ -1186,23 +1213,50 @@
                 setValue('other-meds-quantity.error','');
             </restrictByRole>
 
-
                 var encounterDate = new Date((getValue('encounterDate.value')));
                 var appointment = new Date((jq("#returndate").find('input[type=hidden]').val()));
-                var transferOutDate = new Date((getValue('transfer_out_date.value')));
+                var transferOutDate = new Date((getValue('transfer_out_date.value'))); 
+                var transferOut   = jq('.transfer_out').find("input[type$='checkbox']:checked").val();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1219]: @semujju Is it possible to give the checkbox an ID - this is more robust

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 4]
```diff
@@ -1186,23 +1213,50 @@
...
+                var returnDate    = getValue('returndate.value');
+                var returnDatePoc = jq('.return-date-poc').find("input[type$='text']").val();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1221]: @semujju Can you give this field an ID as that is a more robust implementation

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 2 above)
Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1218]: @ssmusoke Let me clean this out

**Comment by ssmusoke** [comment]
@semujju I can save the POC form without

**Comment by solemabrothers** [comment]
> @semujju I can save the POC form without

@ssmusoke that should be possible provided that there are no medications prescribed, Remember we have cases when we send clients for other services such as Lab and HTS

**Comment by ssmusoke** [comment]
@solemabrothers Got it - I remembered the discussion in the office

**Pull request merged**